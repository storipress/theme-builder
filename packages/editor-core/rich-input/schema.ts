export type Nodes = 'doc' | 'paragraph' | 'text'
export type Marks = 'link' | 'em' | 'strong' | 'u'

export { richInputExtensions } from '@storipress/tiptap-schema'
