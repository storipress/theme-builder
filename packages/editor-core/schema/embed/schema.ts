export { EmbedSchema } from '@storipress/tiptap-schema'
