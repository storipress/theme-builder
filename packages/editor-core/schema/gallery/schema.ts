export { GallerySchema } from '@storipress/tiptap-schema'
