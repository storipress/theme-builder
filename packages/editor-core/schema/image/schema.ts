export { ImageSchema } from '@storipress/tiptap-schema'
