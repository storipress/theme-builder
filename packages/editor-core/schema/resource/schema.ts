export { ResourceSchema } from '@storipress/tiptap-schema'
