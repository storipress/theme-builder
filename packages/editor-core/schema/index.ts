export { baseExtensions as schemaExtensions } from './extensions'
export { warnExceedSizeLimit } from './schema-bus'
export { render, schema } from './schema-only'
