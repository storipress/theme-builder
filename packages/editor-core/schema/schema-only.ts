import { getSchema } from '@tiptap/core'
import { DOMSerializer } from 'prosemirror-model'

export { richInputExtensions } from '../rich-input/schema'
export { loadDocument, render, renderFragment, schema, schemaExtensions } from '@storipress/tiptap-schema'
export { DOMSerializer, getSchema }
