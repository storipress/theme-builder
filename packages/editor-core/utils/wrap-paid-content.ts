export { wrapPaidContent } from '@storipress/tiptap-schema'
