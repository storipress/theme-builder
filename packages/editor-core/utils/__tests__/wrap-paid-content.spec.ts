import { Node } from 'prosemirror-model'
import { describe, expect, it } from 'vitest'
import { renderFragment, schema } from '../../schema/schema-only'
import { wrapPaidContent } from '../wrap-paid-content'

describe('wrapPaidContent', () => {
  it('wrap paragraph to paidContent', () => {
    const doc = Node.fromJSON(schema, {
      type: 'doc',
      content: [
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'free content 1',
            },
          ],
        },
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'free content 2',
            },
          ],
        },
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'free content 3',
            },
          ],
        },
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'paid content 1',
            },
          ],
        },
      ],
    })

    const res = wrapPaidContent({
      doc,
      schema,
    })

    // 3 free + 1 paid wrapper
    expect(res.childCount).toBe(4)
    expect(renderFragment(res.content)).toMatchSnapshot()
  })

  it('flatten exist paidContent', () => {
    const doc = Node.fromJSON(schema, {
      type: 'doc',
      content: [
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'free content 1',
            },
          ],
        },
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'free content 2',
            },
          ],
        },
        // currently paidContent is only for internal use, so we flatten it
        {
          type: 'paidContent',
          content: [
            {
              type: 'paragraph',
              content: [
                {
                  type: 'text',
                  text: 'free content 3',
                },
              ],
            },
            {
              type: 'paragraph',
              content: [
                {
                  type: 'text',
                  text: 'paid content 1',
                },
              ],
            },
          ],
        },
      ],
    })

    const res = wrapPaidContent({
      doc,
      schema,
    })

    // 3 free + 1 paid wrapper
    expect(res.childCount).toBe(4)
    expect(renderFragment(res.content)).toMatchSnapshot()
  })
})
