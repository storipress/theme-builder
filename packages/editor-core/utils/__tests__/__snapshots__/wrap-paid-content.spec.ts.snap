// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`wrapPaidContent > flatten exist paidContent 1`] = `"<p>free content 1</p><p>free content 2</p><p>free content 3</p><div data-format="paid-content"><p>paid content 1</p></div>"`;

exports[`wrapPaidContent > wrap paragraph to paidContent 1`] = `"<p>free content 1</p><p>free content 2</p><p>free content 3</p><div data-format="paid-content"><p>paid content 1</p></div>"`;
